scripts/generate-compatibility-json.py

```diff
@@ -0,0 +1,329 @@
+                              args.relayer,
+                              args.image)
+
+    # compatibility_json is the json object that will be used as the input to a github workflow
+    # which will expand out into a matrix of tests to run.
+    compatibility_json = {
+        "include": include_entries,
+    }
+    _validate(compatibility_json)
+
+    # output the json on a single line. This ensures the output is directly passable to a github workflow.
+    print(json.dumps(compatibility_json), end="")
+
+
+def _add_test_entries(include_entries, seen, chain, release_version=None, version=None, entrypoint=None, test=None,
+                      relayer=None,
+                      chain_image=None):
+    """_add_test_entries adds two different test entries to the test_entries list. One for chain-a -> chain-b and one
+    from chain-b -> chain-a. entries are only added if there are no duplicate entries that have already been added."""
+    # ensure we don't add duplicate entries.
+    entry = (release_version, version, test, entrypoint, relayer, chain_image)
+
+    if entry not in seen and chain in ("chain-a", "all"):
+        include_entries.append(
+            {
+                "chain-a": release_version,
+                "chain-b": version,
+                "entrypoint": entrypoint,
+                "test": test,
+                "relayer-type": relayer,
+                "chain-image": chain_image
+            }
+        )
+        seen.add(entry)
+
+    entry = (version, release_version, test, entrypoint, relayer, chain_image)
+    if entry not in seen and chain in ("chain-b", "all"):
+        include_entries.append(
+            {
+                "chain-a": version,
+                "chain-b": release_version,
+                "entrypoint": entrypoint,
+                "test": test,
+                "relayer-type": relayer,
+                "chain-image": chain_image
+            }
+        )
+        seen.add(entry)
+
+
+def _get_tags_to_test(min_version: semver.Version, all_versions: List[semver.Version]):
+    """return all tags that are between the min and max versions"""
+    max_version = max(all_versions)
+    return ["v" + str(v) for v in all_versions if min_version <= v <= max_version]
+
+
+def _validate(compatibility_json: Dict):
+    """validates that the generated compatibility json fields will be valid for a github workflow."""
+    if "include" not in compatibility_json:
+        raise ValueError("no include entries found")
+
+    required_keys = frozenset({"chain-a", "chain-b", "entrypoint", "test", "relayer-type", "chain-image"})
+    for k in required_keys:
+        for item in compatibility_json["include"]:
+            if k not in item:
+                raise ValueError(f"key {k} not found in {item.keys()}")
+
+    if len(compatibility_json["include"]) > 256:
+        # if this error occurs, split out the workflow into two jobs, one for chain-a and one for chain-b
```

Comment:
maybe we can add this info to the error itself so that we don't have to go look at the code?